typespec-tests/src/main/java/com/server/versions/notversioned/NotVersionedAsyncClient.java

```diff
@@ -114,15 +118,17 @@ public Mono<Void> withoutApiVersion() {
      */
     @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Void> withQueryApiVersion() {
+    public Mono<Void> withQueryApiVersion(String apiVersion) {
```

Comment:
One question, for not-versioned, in tsp definition, the `api-version` query param can be any name, it's just like a normal query param?

Reply:
Yeah, for that test of "not-versioned", it intends to say that the service is not versioned, so no parameter (even it is called `api-version`) mean anything related to version, and hence just a normal parameter like others.

But the cadl-ranch test may not be the best case for that because it has "version" in `@service`. (discussed https://github.com/Azure/autorest.java/pull/2488#discussion_r1440009515 -- but anyway I assume Azure service would not do this)